### DISCUSSION TITLE
Asset generating more Assets (v1 plugin)

### DISCUSSION THREAD

**Discussion by walling** [parcel-bundler/parcel#4755] [General]
Hallo, is it possible for a v1 plugin to generate multiple assets? Say that I register `.jpg` and load an image in a JS file:

```js
import img from './img.jpg';
console.log(img); // ["/img.1.jpg", "/img.2.jpg", "/img.3.jpg"] - result from plugin
```

Would it be possible for the plugin to generate the `img.1.jpg`, `img.2.jpg`, `img.3.jpg` files and register those as assets to be packaged?

**Comment by mischnic** [comment]
That plugin could return a JS asset with a default export `asset.code = 'export default [' + imagesToReturn.map(i => asset.addURLDependency(i)).join(',') + ']';'` (after somehow generating the these images from the main one.

**Reply by walling** [reply]
Should I generate the images directly in the `this.options.outDir` folder? And then pass that as argument for `addURLDependency`?

**Reply by mischnic** [reply]
No, put the in some other folder (e.g. `generated` next to your `src` folder).
`asset.addURLDependency` takes a path relative to `asset`, so you might need to do `path.relative(asset.filePath, subImage1Path)` and pass that.
